java/flight/flight-core/src/main/java/org/apache/arrow/flight/FlightInfo.java

```diff
@@ -93,7 +93,6 @@ public FlightInfo(Schema schema, FlightDescriptor descriptor, List<FlightEndpoin
    */
   public FlightInfo(Schema schema, FlightDescriptor descriptor, List<FlightEndpoint> endpoints, long bytes,
                     long records, boolean ordered, IpcOption option) {
-    Objects.requireNonNull(schema);
     Objects.requireNonNull(descriptor);
     Objects.requireNonNull(endpoints);
     MetadataV4UnionChecker.checkForUnion(schema.getFields().iterator(), option.metadataVersion);
```

Comment:
This needs to be made conditional on the schema. (Can we cover this in a unit test?)

Reply:
Thanks for catching, let me look into the unit test.

Reply:
Added a test!